File: SDKs/bitbucket.org/2.0/go/pkg/models/operations/getrepositoriesworkspacereposlugdownloadsfilename.go
package operations

import (
	"openapi/pkg/models/shared"
)

type GetRepositoriesWorkspaceRepoSlugDownloadsFilenamePathParams struct {
	Filename  string `pathParam:"style=simple,explode=false,name=filename"`
	RepoSlug  string `pathParam:"style=simple,explode=false,name=repo_slug"`
	Workspace string `pathParam:"style=simple,explode=false,name=workspace"`
}

type GetRepositoriesWorkspaceRepoSlugDownloadsFilenameSecurity struct {
	Oauth2 *shared.SchemeOauth2 `security:"scheme,type=oauth2"`
	Basic  *shared.SchemeBasic  `security:"scheme,type=http,subtype=basic"`
	APIKey *shared.SchemeAPIKey `security:"scheme,type=apiKey,subtype=header"`
}

type GetRepositoriesWorkspaceRepoSlugDownloadsFilenameRequest struct {
	PathParams GetRepositoriesWorkspaceRepoSlugDownloadsFilenamePathParams
	Security   GetRepositoriesWorkspaceRepoSlugDownloadsFilenameSecurity
}

type GetRepositoriesWorkspaceRepoSlugDownloadsFilenameResponse struct {
	ContentType string
	StatusCode  int64
	Error       map[string]interface{}
}

File: SDKs/bitbucket.org/2.0/python/src/sdk/models/operations/get_repositories_workspace_repo_slug_downloads_filename_.py
from dataclasses import dataclass, field
from typing import Any,Optional


@dataclass
class GetRepositoriesWorkspaceRepoSlugDownloadsFilenamePathParams:
    filename: str = field(default=None, metadata={'path_param': { 'field_name': 'filename', 'style': 'simple', 'explode': False }})
    repo_slug: str = field(default=None, metadata={'path_param': { 'field_name': 'repo_slug', 'style': 'simple', 'explode': False }})
    workspace: str = field(default=None, metadata={'path_param': { 'field_name': 'workspace', 'style': 'simple', 'explode': False }})
    

@dataclass
class GetRepositoriesWorkspaceRepoSlugDownloadsFilenameSecurityOption1:
    oauth2: shared.SchemeOauth2 = field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclass
class GetRepositoriesWorkspaceRepoSlugDownloadsFilenameSecurityOption2:
    basic: shared.SchemeBasic = field(default=None, metadata={'security': { 'scheme': True, 'type': 'http', 'sub_type': 'basic' }})
    

@dataclass
class GetRepositoriesWorkspaceRepoSlugDownloadsFilenameSecurityOption3:
    api_key: shared.SchemeAPIKey = field(default=None, metadata={'security': { 'scheme': True, 'type': 'apiKey', 'sub_type': 'header' }})
    

@dataclass
class GetRepositoriesWorkspaceRepoSlugDownloadsFilenameSecurity:
    option1: Optional[GetRepositoriesWorkspaceRepoSlugDownloadsFilenameSecurityOption1] = field(default=None, metadata={'security': { 'option': True }})
    option2: Optional[GetRepositoriesWorkspaceRepoSlugDownloadsFilenameSecurityOption2] = field(default=None, metadata={'security': { 'option': True }})
    option3: Optional[GetRepositoriesWorkspaceRepoSlugDownloadsFilenameSecurityOption3] = field(default=None, metadata={'security': { 'option': True }})
    

@dataclass
class GetRepositoriesWorkspaceRepoSlugDownloadsFilenameRequest:
    path_params: GetRepositoriesWorkspaceRepoSlugDownloadsFilenamePathParams = field(default=None)
    security: GetRepositoriesWorkspaceRepoSlugDownloadsFilenameSecurity = field(default=None)
    

@dataclass
class GetRepositoriesWorkspaceRepoSlugDownloadsFilenameResponse:
    content_type: str = field(default=None)
    status_code: int = field(default=None)
    error: Optional[dict[str, Any]] = field(default=None)
    
